src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -180,8 +205,30 @@ private static int VerifyCertChain(IntPtr storeCtxPtr, IntPtr arg)
 
                         using (X509Certificate2 leafCert = new X509Certificate2(leafCertPtr))
                         {
-                            success = chain.Build(leafCert);
-                            AddChannelBindingToken(leafCert, arg);
+                            var serverCallback = easy._handler._serverCertificateValidationCallback;
+                            if (serverCallback == null)
+                            {
+                                SslPolicyErrors errors = CertificateValidation.BuildChainAndVerifyProperties(
+                                    chain, leafCert, checkCertName: false, hostName: null);
+                                success = errors == SslPolicyErrors.None;
```

Comment:
Ok, thanks, I'm going to change it to always do the name check, not just when the callback is being used (which is the only time the SslPolicyErrors actually matters, otherwise it's an internal implementation detail).